src/publisher.ts

```diff
@@ -28,35 +28,46 @@ export class Publisher {
       console.warn('name should not contain ":"');
       name = name.replace(':', ' ');
     }
-    this.id = Math.round(Math.random() * 1000000) + '';
+    this.id = String(Math.round(Math.random() * 1000000));
   }
 
-  start(callback?: Function) {
-    if (this.running) {
-      return;
-    }
-    this.running = true;
-
-    const client = this.client = dgram.createSocket('udp4');
-    client.on('listening', () => {
-      client.setBroadcast(true);
-      this.timer = setInterval(this.sayHello.bind(this), this.interval);
-      this.sayHello();
-      callback && callback();
+  start(): Promise<void> {
+    return new Promise<void>((resolve, reject) => {
+      if (this.running) {
+        return;
+      }
+      this.running = true;
+
+      const client = this.client = dgram.createSocket('udp4');
+      client.on('listening', () => {
+        client.setBroadcast(true);
+        this.timer = setInterval(this.sayHello.bind(this), this.interval);
+        this.sayHello().then(() => {
```

Comment:
I am not sure a failing "sayHello()" should make start() to fail. probably listening to "error" is a better idea. Like start should only fail if the socket can't be opened.